src/lib.rs

```diff
@@ -357,21 +357,22 @@ impl RequestBuildah for DefaultRequestBuilder {
         self
     }
 
-    fn send(self) -> std::result::Result<Vec<u8>, Error> {
+    fn send(mut self) -> std::result::Result<Self::ReturnValue, Error> {
         let mut response = self.inner.send()?;
         if response.status() != StatusCode::OK {
             return Err(Error::HttpStatus(response.status()));
         }
-        let mut buf = vec![];
-        let _ = response.read_to_end(&mut buf)?;
+        let mut buf = String::with_capacity(200);
+        let _ = response.read_to_string(&mut buf)?;
         Ok(buf)
     }
 }
 
 pub trait RequestBuildah {
-    type Error: std::error::Error + Debug;
+    type Error: Debug;
+    type ReturnValue;
 
-    fn new(method: http::Method, url: impl IntoUrl) -> Self;
+    fn new(method: http::Method, url: &'_ str) -> Self;
```

Comment:
I'm not very happy with this signature, as to achieve connection pooling it requires you to use lazy_static type of solution. Ideally I would like to pass here a `reqwest::Client` type of variable to clone it for reusing the connection pool. I think adding a third associated type `ClientBuilder` or sth would be a good idea.